resin/models/build.py

```diff
@@ -0,0 +1,79 @@
+from ..base_request import BaseRequest
+from ..settings import Settings
+from ..token import Token
+from .. import exceptions
+
+
+class Build(object):
+    """
+    This class implements model for build for Resin Python SDK.
+
+    """
+
+    def __init__(self):
+        self.base_request = BaseRequest()
+        self.settings = Settings()
+        self.token = Token()
+
+    def get(self, id):
+        """
+        Get a specific build.
+
+        Args:
+            id (str): build id.
+
+        Returns:
+            dict: build info.
+
+        Raises:
+            BuildNotFound: if build couldn't be found.
+
+        """
+
+        params = {
+            'filter': 'id',
+            'eq': id
+        }
+        build = self.base_request.request(
+            'build', 'GET', params=params,
+            endpoint=self.settings.get('pine_endpoint')
+        )
+        if build:
+            return build
+        else:
+            raise exceptions.BuildNotFound(id)
+
+    def get_all_by_application(self, app_id):
+        """
+        Get a specific build.
+
+        Args:
+            app_id (str): application id.
+
+        Returns:
+            list: list of build info.
+
+        Raises:
+            BuildNotFound: if build couldn't be found.
+
+        """
+
+        params = {
+            'filter': 'application',
+            'eq': app_id
+        }
+        full_builds = self.base_request.request(
+            'build', 'GET', params=params,
+            endpoint=self.settings.get('pine_endpoint')
+        )
+        builds = []
+        for i in full_builds['d']:
+            builds.append({
+                k: i[k] for k in (
+                    'id', 'created_at', 'status',
+                    'push_timestamp', 'end_timestamp',
+                    'start_timestamp', 'project_type',
+                    'commit_hash', 'message', 'log'
```

Comment:
Ah, no, you've misunderstood the JS there. What that code is trying to do (I think) is make `.get()` return everything, and `.getAllByApplication` return everything except the `log` property, because the full build logs can be huge. Here you've included `log` (note that it's explicitly commented out in the JS code), which means this code doesn't really do anything, and means that this method will be much slower than it needs to be (since it downloads all the logs for every build ever for an application).

I think you want to change the code in this loop, to instead take the build object, and just remove the the `log` property. That doesn't really solve the problem that the JS code has solved, because you'll still have to download all the logs before you get to this line, but it lets us solve it later. It'd be better to pass proper pine options if you can, but if you can't at least this way users won't start depending on this method returning builds with `.log`, and we can stop transferring them entirely once you've added more pine option support.